### Part 2-2  Using airbnb's property location data to determine the number of apartments near subway stations


In [1]:
import pandas as pd
from geopy.distance import geodesic

In [2]:
df0 = pd.read_csv('listings.csv',encoding='UTF-8')

In [3]:
df0.shape

(37012, 74)

In [4]:
df0.columns

Index(['id', 'listing_url', 'scrape_id', 'last_scraped', 'name', 'description',
       'neighborhood_overview', 'picture_url', 'host_id', 'host_url',
       'host_name', 'host_since', 'host_location', 'host_about',
       'host_response_time', 'host_response_rate', 'host_acceptance_rate',
       'host_is_superhost', 'host_thumbnail_url', 'host_picture_url',
       'host_neighbourhood', 'host_listings_count',
       'host_total_listings_count', 'host_verifications',
       'host_has_profile_pic', 'host_identity_verified', 'neighbourhood',
       'neighbourhood_cleansed', 'neighbourhood_group_cleansed', 'latitude',
       'longitude', 'property_type', 'room_type', 'accommodates', 'bathrooms',
       'bathrooms_text', 'bedrooms', 'beds', 'amenities', 'price',
       'minimum_nights', 'maximum_nights', 'minimum_minimum_nights',
       'maximum_minimum_nights', 'minimum_maximum_nights',
       'maximum_maximum_nights', 'minimum_nights_avg_ntm',
       'maximum_nights_avg_ntm', 'calendar_upd

### Limit the data to properties that meet the following criteria
- Property_type: Entire apartment or Entire condominium or Entire serviced apartment
- review_scores_rating >= 95
- maximum_nights >= 90

In [5]:
df0= df0[(df0['review_scores_rating'] >= 95) & (df0['maximum_nights'] >= 90)]

In [6]:
df0.shape

(12043, 74)

In [7]:
list = ['Entire apartment','Entire condominium','Entire serviced apartment']
df = df0[df0.property_type.isin(list)]

In [8]:
df.shape

(6036, 74)

#### The number of apartments that satisfy the conditions is 6036.

In [9]:
st0 = pd.read_csv('DOITT_SUBWAY_STATION_01_13SEPT2010.csv',encoding='UTF-8')

In [10]:
st0.head()

URL  OBJECTID              NAME  \
0  http://web.mta.info/nyct/service/         1          Astor Pl   
1  http://web.mta.info/nyct/service/         2          Canal St   
2  http://web.mta.info/nyct/service/         3           50th St   
3  http://web.mta.info/nyct/service/         4         Bergen St   
4  http://web.mta.info/nyct/service/         5  Pennsylvania Ave   

                                       the_geom           LINE  \
0  POINT (-73.99106999861966 40.73005400028978)  4-6-6 Express   
1  POINT (-74.00019299927328 40.71880300107709)  4-6-6 Express   
2  POINT (-73.98384899986625 40.76172799961419)            1-2   
3  POINT (-73.97499915116808 40.68086213682956)          2-3-4   
4  POINT (-73.89488591154061 40.66471445143568)            3-4   

                                               NOTES  
0  4 nights, 6-all times, 6 Express-weekdays AM s...  
1  4 nights, 6-all times, 6 Express-weekdays AM s...  
2                              1-all times, 2-nights  
3           4-nights, 3-all other times, 2-all times  
4                        4-nights, 3-all other times

In [11]:
st0.drop(columns=['URL','OBJECTID','NOTES'],inplace=True)
st = st0.rename(columns={'NAME': 'Station_Name'})

In [12]:
  st['lat'] = 0.0
  st['lng'] = 0.0
  st['numuni'] = 0

for j in range(len(st)):
   nam = st['Station_Name'][j] 
   p = st['the_geom'][j]
   x = float(p.split(" ")[1].strip('('))
   y = float(p.split(" ")[2].rstrip(')'))
   st.loc[st.Station_Name == nam,'lat'] = y
   st.loc[st.Station_Name == nam,'lng'] = x


In [13]:
st.drop(columns=['the_geom'],inplace=True)
st.head()

Station_Name           LINE        lat        lng  numuni
0          Astor Pl  4-6-6 Express  40.730054 -73.991070       0
1          Canal St  4-6-6 Express  40.718814 -74.001055       0
2           50th St            1-2  40.762456 -73.985984       0
3         Bergen St          2-3-4  40.686111 -73.990756       0
4  Pennsylvania Ave            3-4  40.664714 -73.894886       0

In [14]:
st.iloc[1,4]

0

#### Using the location information of the apartments that meet the conditions and <br>the location information of the subway station, count the number of apartments <br>within 300 meters of the station.

In [15]:
for i in range (len(st)):
  nol = 0
  stp = (st.iloc[i].lat,st.iloc[i].lng)
  for j in range(len(df)):
     bnbp =(df.iloc[j].latitude,df.iloc[j].longitude)
     dis = geodesic(stp,bnbp).m
     #print(i,j,dis)
     if dis <= 300 :
         nol = nol + 1
  print('***',i,nol)
  st.iloc[i,4] = nol


*** 0 27
*** 1 17
*** 2 46
*** 3 30
*** 4 0
*** 5 0
*** 6 13
*** 7 12
*** 8 2
*** 9 3
*** 10 43
*** 11 6
*** 12 0
*** 13 0
*** 14 11
*** 15 0
*** 16 2
*** 17 1
*** 18 1
*** 19 0
*** 20 0
*** 21 1
*** 22 0
*** 23 0
*** 24 5
*** 25 0
*** 26 1
*** 27 0
*** 28 2
*** 29 0
*** 30 33
*** 31 0
*** 32 23
*** 33 2
*** 34 4
*** 35 18
*** 36 1
*** 37 1
*** 38 0
*** 39 1
*** 40 1
*** 41 11
*** 42 0
*** 43 0
*** 44 5
*** 45 6
*** 46 0
*** 47 0
*** 48 19
*** 49 2
*** 50 0
*** 51 0
*** 52 1
*** 53 0
*** 54 0
*** 55 36
*** 56 23
*** 57 6
*** 58 0
*** 59 8
*** 60 14
*** 61 0
*** 62 0
*** 63 0
*** 64 0
*** 65 5
*** 66 1
*** 67 0
*** 68 1
*** 69 18
*** 70 9
*** 71 0
*** 72 0
*** 73 1
*** 74 4
*** 75 0
*** 76 54
*** 77 7
*** 78 9
*** 79 2
*** 80 33
*** 81 5
*** 82 0
*** 83 48
*** 84 9
*** 85 37
*** 86 0
*** 87 13
*** 88 0
*** 89 17
*** 90 0
*** 91 26
*** 92 5
*** 93 21
*** 94 4
*** 95 26
*** 96 28
*** 97 0
*** 98 0
*** 99 1
*** 100 8
*** 101 16
*** 102 10
*** 103 1
*** 104 15
*** 105 26
*** 106 4
*** 107 2

In [16]:
st.head()

Station_Name           LINE        lat        lng  numuni
0          Astor Pl  4-6-6 Express  40.730054 -73.991070      27
1          Canal St  4-6-6 Express  40.718814 -74.001055      17
2           50th St            1-2  40.762456 -73.985984      46
3         Bergen St          2-3-4  40.686111 -73.990756      30
4  Pennsylvania Ave            3-4  40.664714 -73.894886       0

In [17]:
st.tail()

Station_Name         LINE        lat        lng  numuni
468  Coney Island - Stillwell Av      D-F-N-Q  40.577281 -73.981236       0
469       34th St - Hudson Yards  7-7 Express  40.755446 -74.002197       3
470                      72nd St            Q  40.768803 -73.958362      36
471                      86th St            Q  40.777861 -73.951771      37
472                      96th St            Q  40.784237 -73.947066      23

In [18]:
st.to_csv('station.csv')